subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy

```diff
@@ -52,6 +78,84 @@ class JUnitPlatformIntegrationTest extends JUnitPlatformIntegrationSpec {
         succeeds('test')
     }
 
+    def 'can handle dynamically created test'() {
+        given:
+        buildFile << """
+        dependencies {
+            testImplementation 'org.junit.platform:junit-platform-runner:1.0.3'
+        }
+        """
+        createDynamicJupiterTest()
+
+        when:
+           def build = executer
+                .withTasks('test')
+                .withTestConsoleAttached()
+                .withConsole(console)
+                .start()
+
+        build.waitForFailure()
+        def output = LogContent.of(build.standardOutput).ansiCharsToColorText().withNormalizedEol()
+
+        if (output.contains("dynamicTestStream()[3]")) {
+            throw new AssertionError("expected displayName for dynamic test but found array method instead");
+        }
+        if (!output.contains("is even")) {
+            throw new AssertionError("expected displayName for dynamic test but not present");
+        }
+
+        then:
+
+        new DefaultTestExecutionResult(testDirectory)
+            .assertTestClassesExecuted('org.gradle.JUnitJupiterDynamicTest')
+            .testClass('org.gradle.JUnitJupiterDynamicTest')
+            .assertTestCount(3, 1, 0)
+
+        where:
+        console << [ConsoleOutput.Plain] //, ConsoleOutput.Rich, ConsoleOutput.Verbose
+    }
+
+    def 'can handle dynamically created test on log info'() {
```

Comment:
Both tests looks same excluding passing --info option. I'm not sure that both tests are needed (we could discuss) but if it is then we could reduce this duplication with Spock's data pipes